core/core_integrations/slack.js

```diff
@@ -17,596 +15,553 @@ inTransaction = {},
+    }
+    for (var j = 0; j < data.groups.length; j++) {
+        channels[data.groups[j].id] = data.groups[j];
+    }
+    for (var k = 0; k < data.ims.length; k++) {
+        channels[data.ims[k].id] = data.ims[k];
+    }
 
-sendPing = teamId => {
-	let slackTeams = exports.config.slack_teams,
-		team = slackTeams[teamId],
-		pongId = ++team.pong_id;
-
-	if (!pongId) {
-		pongId = 1;
-		team.pong_id = pongId;
-	}
-
-	let body = {
-		'id': team.pong_id,
-		'type':'ping',
-		'timestamp': Date.now()
-	};
-	body = JSON.stringify(body);
-	sendSocketMessage(teamId, body);
-
-	tempTeamData[teamId].pongTimeout = setTimeout(() => {
-		if (team.pong_id === pongId && !shuttingDown) {
-			console.debug("slack-> terminating connection as socket failed to respond to ping request.");
-			sockets[teamId].close();
-		}
-	}, defaultPingResponseTimeout);
+    return channels;
 },
 
-pong = function(pong, teamId) {
-	console.debug("slack-> recieved pong");
-	exports.config.slack_teams[teamId].timeSincelastMessageRecieved = Date.now();
-	exports.config.slack_teams[teamId].pong_id++;
+generateUserMap = function(data) {
+    var map = {};
+    for (var i = 0; i < data.users.length; i++) {
+        map[data.users[i].id] = data.users[i];
+    }
+    return map;
 },
 
-findSlackBot = function(body, slackTeam) {
-	for (var i = 0; i < body.users.length; i++) {
-		if (body.users[i].name === 'slackbot') {
-			slackTeam.bot_id = body.users[i].id;
-			break;
-		}
-	}
+init = function(token, callback) {
+    var body = {
+        token: token,
+        mpim_aware: true
+    };
+    request({
+        uri: 'https://slack.com/api/rtm.start',
+        method: 'GET',
+        qs: body
+    },
+    function (error, response, body) {
+        if (error) {
+            console.debug('slack-> failed to Initialise connection: ' + error);
+            callback(false);
+            return;
+        }
+        var teamId = null,
+        team;
+
+        try {
+            body = JSON.parse(body);
+        }
+        catch (e) {
+            console.debug('slack-> failed to parse message: ' + body);
+            console.critical(e);
+            callback(false);
+            return;
+        }
+
+        if (response.statusCode !== 200) {
+            console.debug('slack-> error: ' + response.statusCode);
+            callback(false);
+        }
+        else if (!body.ok) {
+            console.debug('slack-> Failed to connect to slack server with token: ' + token + ' error: ' + body.error);
+            callback(false);
+        }
+        else {
+            teamId = body.team.id;
+            teamData[teamId] = team = body;
+            team.token = token;
+            team.event_id = 1;
+            team.lastMessageSinceConnection = false;
+
+            // Generate user map
+            team.users = generateUserMap(body);
+            team.allChannels = generateChannelsMap(body);
+
+            // find slackbot
+            findSlackBot(body, team);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["findSlackBot" was used before it was defined](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l250)